src/ORM/DataObject.php

```diff
@@ -1219,7 +1221,7 @@ public function merge($rightObj, $priority = 'right', $includeRelations = true,
                     $leftComponents = $leftObj->getComponents($relationship);
                     $rightComponents = $rightObj->getComponents($relationship);
                     if ($rightComponents && $rightComponents->exists()) {
-                        $leftComponents->addMany($rightComponents->column('ID'));
+                        $leftComponents->addMany($rightComponents->getIDList());
```

Comment:
This loop just adds items to a `DataList`.
The order in which items are added doesn't matter.